crates/interpreter/src/instruction_result.rs

```diff
@@ -4,68 +4,91 @@ use crate::primitives::{HaltReason, OutOfGasError, SuccessReason};
 #[derive(Clone, Copy, Debug, Default, PartialEq, Eq, Hash)]
 #[cfg_attr(feature = "serde", derive(serde::Serialize, serde::Deserialize))]
 pub enum InstructionResult {
-    // success codes
+    // Success Codes
     #[default]
+    /// Execution should continue to the next one.
     Continue = 0x00,
+    /// Halt execution.
```

Comment:
`Halt` can mean different behaviours
```suggestion
    /// Encountered a `STOP` opcode
```
Not sure how to phase this one

Reply:
I used the description shown in [evm.codes](https://www.evm.codes/#00?fork=cancun) but happy to change it.

<img width="1403" alt="Screenshot 2024-08-06 at 13 48 42" src="https://github.com/user-attachments/assets/44700fdf-00b6-47d5-b8f6-941704cf9bab">

Reply:
Calling it halt is not correct, as STOP is considered valid/successful execution. Lets change it.